3rdparty/CMakeLists.txt

```diff
@@ -0,0 +1,291 @@
+find_package(PkgConfig)
+
+if(APPLE)
+	set(PLATFORM_ARCH "macosx/x86_64")
+elseif(WIN32)
+	set(PLATFORM_ARCH "Windows/x86_64")
+else()
+	set(PLATFORM_ARCH "linux/x86_64")
+endif()
+
+# Dummy target to use when lib isn't available
+add_library(3rdparty_dummy_lib INTERFACE)
+
+# ZLib
+find_package(ZLIB QUIET)
+if (NOT ZLIB_FOUND)
+	add_subdirectory(3rdparty/zlib EXCLUDE_FROM_ALL)
+	set(ZLIB_INCLUDE_DIR "${CMAKE_CURRENT_SOURCE_DIR}/zlib" "${CMAKE_CURRENT_BINARY_DIR}/zlib")
+	set(ZLIB_LIBRARY zlibstatic)
+endif()
+
+add_library(3rdparty_zlib INTERFACE)
+target_link_libraries(3rdparty_zlib INTERFACE ${ZLIB_LIBRARY})
+target_include_directories(3rdparty_zlib INTERFACE ${ZLIB_INCLUDE_DIR})
+
+# libPNG
+# Select the version of libpng to use, default is builtin
+if (NOT USE_SYSTEM_LIBPNG)
+	# We use libpng's static library and don't need to build the shared library and run the tests
+	set(PNG_SHARED OFF CACHE BOOL "Build shared lib" FORCE)
+	set(PNG_TESTS OFF CACHE BOOL "Build libpng tests" FORCE)
+	set(SKIP_INSTALL_ALL ON)
+	add_subdirectory(libpng EXCLUDE_FROM_ALL)
+
+	set(LIBPNG_TARGET png_static)
+else()
+	find_package(PNG REQUIRED)
+
+	add_library(3rdparty_system_libpng INTERFACE)
+	target_include_directories(3rdparty_system_libpng INTERFACE ${PNG_INCLUDE_DIR})
+	target_link_libraries(3rdparty_system_libpng INTERFACE ${PNG_LIBRARY})
+	target_compile_definitions(3rdparty_system_libpng INTERFACE ${PNG_DEFINITIONS})
+
+	set(LIBPNG_TARGET 3rdparty_system_libpng)
+endif()
+
+
+# pugixml
+add_subdirectory(pugixml EXCLUDE_FROM_ALL)
+
+
+# hidapi
+add_library(3rdparty_hidapi INTERFACE)
+target_include_directories(3rdparty_hidapi INTERFACE hidapi/hidapi)
+
+if(APPLE)
+	add_subdirectory(hidapi/mac EXCLUDE_FROM_ALL)
+	target_include_directories(hidapi-mac PUBLIC hidapi/hidapi)
+
+	target_link_libraries(3rdparty_hidapi INTERFACE hidapi-mac "-framework CoreFoundation" "-framework IOKit")
+elseif(CMAKE_SYSTEM MATCHES "Linux")
+	add_subdirectory(hidapi/linux EXCLUDE_FROM_ALL)
+	target_include_directories(hidapi-hidraw PUBLIC hidapi/hidapi)
+
+	target_link_libraries(3rdparty_hidapi INTERFACE hidapi-hidraw udev)
+elseif(MSVC)
+	add_subdirectory(hidapi/windows EXCLUDE_FROM_ALL)
+	target_include_directories(hidapi-hid PUBLIC hidapi/hidapi)
+
+	target_link_libraries(3rdparty_hidapi INTERFACE hidapi-hid Shlwapi.lib)
+else()
+	add_subdirectory(hidapi/libusb EXCLUDE_FROM_ALL)
+	target_link_libraries(3rdparty_hidapi INTERFACE hidapi-libusb usb)
+endif()
+
+
+# yaml-cpp
+set(YAML_CPP_BUILD_TESTS OFF CACHE BOOL "Enable testing" FORCE)
+set(YAML_CPP_BUILD_TOOLS OFF CACHE BOOL "Enable parse tools" FORCE)
+set(YAML_CPP_BUILD_CONTRIB OFF CACHE BOOL "Enable contrib stuff in library" FORCE)
+add_subdirectory(yaml-cpp EXCLUDE_FROM_ALL)
+
+
+# xxHash
+set(XXHASH_BUNDLED_MODE ON)
+set(BUILD_SHARED_LIBS OFF CACHE BOOL "Make xxHash build static libs")
+add_subdirectory(xxHash/cmake_unofficial)
```

Comment:
I know it wasn't in the original CMake, but while you're changing it, could you make this EXCLUDE_FROM_ALL

Reply:
Added EXCLUDE_FROM_ALL